services/app-api/src/postgres/contractAndRates/parseContractWithHistory.ts

```diff
@@ -132,19 +135,41 @@ function contractWithHistoryToDomainModel(
                 )
             }
 
-            const draftPrismaRates = contractRev.draftRates
-
-            draftRates = draftPrismaRates.map((r) => {
-                return {
-                    id: r.id,
-                    createdAt: r.createdAt,
-                    updatedAt: r.updatedAt,
-                    status: getContractRateStatus(r.revisions),
-                    stateCode: r.stateCode,
-                    stateNumber: r.stateNumber,
-                    revisions: [],
-                }
-            })
+            // if we have a draft revision, we should set draftRates
+            const draftRatesOrError = contract.draftRates.map((dr) =>
+                rateWithHistoryToDomainModel(dr.rate)
+            )
+            const firstError: Error | undefined = draftRatesOrError.find(
+                (dr): dr is Error => dr instanceof Error
+            )
+            if (firstError) {
+                return firstError
+            } else {
+                const allDraftRates: RateType[] =
+                    draftRatesOrError as RateType[]
```

Comment:
Do we really intend to do this type coercion where an Error is deemed Rate Type list?